Common/Python/PandasData.cs

```diff
@@ -63,8 +65,15 @@ public PandasData(IBaseData baseData)
                 // and add the field named 'value' since it is the reference value
                 columns = "value," + string.Join(",", ((DynamicData)baseData).GetStorageDictionary().Keys);
             }
-            
-            _series = columns.Split(',').ToDictionary(k => k, v => new List<object>());
+            // C# custom data
+            else if (baseData is BaseData)
+            {
+                _type = baseData.GetType();
+                var properties = _type.GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
```

Comment:
Won't `DeclaredOnly` cause issues with subclasses? From [MSDN](https://msdn.microsoft.com/en-us/library/system.reflection.bindingflags(v=vs.110).aspx):

```
Specifies that only members declared at the level of the supplied type's hierarchy should be considered. Inherited members are not considered.
```

If we don't have an explicit reason to use `BindingFlags.DeclaredOnly` then I would recommend removing it from this call. Also, might want to test this out. Consider the following class structure:

```csharp
public class MyCustomDataBaseClass : BaseData
{
    public int MyCustomDataProperty { get; set; }
}

public class MyDerivedCustomDataClass : MyCustomDataBaseClass
{
    public int MyDerivedCustomDataProperty { get; set; }
}
```

I think with what you've provided we won't get access to the `MyCustomDataProperty`

Reply:
That is the idea: we don't want the subclass properties. We just want `EndTime`, `Value` and those there were defined/declared in the custom class.

Reply:
Wait... why don't we want subclass properties? I think you're saying we only want `EndTime` and `Value` from `BaseData`... but we should get all other sub class properties for the exact case described above. It reasonable to think that the user would want the `MyCustomDataBaseClass.MyCustomDataProperty` to be in the data frame, right?

A more tangible example perhaps is when a user subclasses `TradeBar` to get the OHLC properties for free. With this example, I think we would miss out on those properties given a type:
```csharp
public class ExtendedTradeBar : TradeBar
{
    public decimal TheAnswerToLife { get; set; } = 42;
}
```

I think with the proposed solution, only `TheAnswerToLife` would end up in the frame (along with `EndTime`, and `Value`) but I would also expect `Open`, `High`, `Low,` and `Close` from `TradeBar` as well.

Reply:
I am not accounting for subclassing `TradeBar`.
```csharp
else if (baseData is BaseData)
```
Maybe I should do the other way around: remove the undesired properties like `DataType`, `Symbol`, etc...

Reply:
maybe something like this would do the trick:
```csharp
var properties = type.GetProperties()
    .Where(p => p.DeclaringType != typeof (BaseData)
             && p.DeclaringType != typeof (IBaseData));
```

And then manually add `EndTime` and `Value` as before.